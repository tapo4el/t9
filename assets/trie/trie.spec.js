import Trie from './trie';

describe('Trie', () => {
    let trie;

    beforeEach(() => {
        trie = new Trie();
    });

    it ('should initialize correctly', () => {
        expect(trie).toMatchSnapshot();
    });

    it ('insert work correctly and to be case insensitive', () => {
        trie.insert('ab');
        trie.insert('DE');
        expect(trie).toMatchSnapshot();
    });

    it ('getSuggestions work correctly with empty input string', () => {
        trie.insert('aaa');
        trie.insert('ddd');
        expect(trie.getSuggestions('')).toEqual([]);
    });

    it ('getSuggestions work correctly', () => {
        trie.insert('aaa');
        trie.insert('ddd');
        expect(trie.getSuggestions('2')).toEqual(['aaa']);
        expect(trie.getSuggestions('4')).toEqual([]);
    });
});
