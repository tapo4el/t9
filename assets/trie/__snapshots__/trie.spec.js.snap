// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trie insert work correctly and to be case insensitive 1`] = `
Trie {
  "root": Node {
    "children": Object {
      "2": Node {
        "children": Object {
          "2": Node {
            "children": Object {},
            "words": Array [
              "ab",
            ],
          },
        },
        "words": Array [],
      },
      "3": Node {
        "children": Object {
          "3": Node {
            "children": Object {},
            "words": Array [
              "de",
            ],
          },
        },
        "words": Array [],
      },
    },
    "words": Array [],
  },
}
`;

exports[`Trie should initialize correctly 1`] = `
Trie {
  "root": Node {
    "children": Object {},
    "words": Array [],
  },
}
`;
